iOS10-HW11-Bessonov Ilia/Sources/ViewController.swift

```diff
@@ -0,0 +1,360 @@
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.imageView?.translatesAutoresizingMaskIntoConstraints = false
+        button.backgroundColor = UIColor.blue.withAlphaComponent(0.6)
+        button.clipsToBounds = true
+        button.layer.cornerRadius = 18
+        button.setImage(UIImage(named: "ic-twitter")!, for: .normal)
+        button.setTitle("Twitter", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 11, weight: .bold)
+        button.contentHorizontalAlignment = .center
+        button.imageView?.trailingAnchor.constraint(equalTo: button.centerXAnchor, constant: -15).isActive = true
+        button.imageView?.centerYAnchor.constraint(equalTo: button.centerYAnchor).isActive = true
+        button.addTarget(self, action: #selector(twitterPressed), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var signUpStack: UIStackView = {
+        let stack = UIStackView()
+        stack.axis = .horizontal
+        stack.alignment = .center
+        stack.distribution = .equalSpacing
+        stack.spacing = 10
+        stack.backgroundColor = .clear
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        stack.layoutMargins = UIEdgeInsets(top: 0, left: 15, bottom: 0, right: 15)
+        stack.isLayoutMarginsRelativeArrangement = true
+        stack.addArrangedSubview(haventAccountLabel)
+        stack.addArrangedSubview(signUpButton)
+        return stack
+        }()
+
+    private lazy var haventAccountLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = "Dont have account?"
+        label.textColor = .systemGray
+        label.font = .systemFont(ofSize: 12, weight: .medium)
+        label.textAlignment = .right
+        label.numberOfLines = 1
+        return label
+    }()
+
+    private lazy var signUpButton: UIButton = {
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.backgroundColor = .clear
+        button.setTitle("Sign up", for: .normal)
+        button.contentHorizontalAlignment = .left
+        button.setTitleColor(UIColor.blue.withAlphaComponent(0.5), for: .normal)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 12, weight: .bold)
+        button.addTarget(self, action: #selector(signUpButtonPressed), for: .touchUpInside)
+        return button
+    }()
+
+    // MARK: - Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupHierarchy()
+        setupLayout()
+    }
+
+    // MARK: - Setup
+
+    private func setupHierarchy() {
+        view.layer.insertSublayer(gradient, at: 0)
+        view.addSubviews([loginLabel, loginTextField, passwordTextField, loginButton, forgotPasswordButton, separateLineStack, socialButtonsStack, signUpStack])
```

Comment:
это можно вот так вот обернуть
```suggestion
 [loginLabel, loginTextField, passwordTextField,
         loginButton, forgotPasswordButton, separateLineStack,
         socialButtonsStack, signUpStack
        ].forEach {
            view.addSubview($0)
        }
```